### ISSUE TITLE
Details Page -- bugs with airport info and METARS

### ISSUE THREAD

**Issue by KayRJay** [ericberman/MyFlightbookWeb#403]
My flight on 8/23/19 included several airports and had telemetry.   For this flight, the "Show current METARS" button in the flight details shows the current METAR for only one airport, but the button is plural.   The button may not be necessary, as there is a link to display the METAR for each airport.

Below the button there is a summary of the flight including intermediate stops.  Very nice.   Mine reads  Average Speed: 498.1kts ● Path distance: 30.1nm ● Route distance: 99.6nm ● Longest Segment: 49.1nm ● Furthest points from each other: KHCR and U14 (49.1nm).

Again, this is only for flights with telemetry, I guess.  It would be helpful to have similar information when no telemetry is available.   There could be approximations of distances flown as the distance between airports based on their GPS coordinates, and possibly speed if date/time of departure and landings are available.

The airports are not listed in the order visited.   So, my initial departure and final arrival airport (the same in this case) are listed twice.   Also, I entered the airport as HCR for departure and KHCR for final arrival.   This should be disambiguated, and each visited airport listed only once. 

Also, for the flight on  flight on 8/21/19, the "Show current METARs" simply refreshes the current page and does not display any METARs.

For the 8/23 flight, the airport is correctly displayed as Heber.  However, clicking on METAR shows the METAR for Transener S.A. Heliport in Argentina.   I entered the airport as HCR, not KHCR.  It's odd that the name is properly displayed but not the METAR.

Although the spelled out name of the airport looks like a link, nothing happens when I click on it.

It's not clear what the red circle with the white double bar indicates.  There's no pop-up to explain.

**Comment by ericberman** [comment]
There's not a lot I can do on this.

The "Show METARS" button and the "METAR" link next to each airport do different things.  

The "Show METARS" hits a web service and displays any results it gets back, inline.  See below for a flight of mine.  This service generally (but not always!) only works for ICAO codes, and I have no way of knowing ahead of time whether or not you'll get any results; the reason I make you click is so that I'm not pounding this (free) service and getting cut off.
![metars](https://user-images.githubusercontent.com/16455286/68344577-3316f300-00a4-11ea-9fb7-493c15b7da17.png)
The "METAR" link next to each airport, on the other hand, links to a 3rd party web page (so you're leaving the site) and can accept a wider array of airport formats, but it's not inline (and I still don't know if it will find the code).  All three of these are working from different databases.  In this case, HCR is not being found on the web service (but "KHCR" does get found).  That's why when you hit refresh, I don't show any results; I have no results to show.  And it's also why HCR maps to Argentina - you've left MyFlightbook and are now on a different web site with a different database.

I use the plural "METARS" (should be METARs perhaps?) since I don't know how many results I'll get back.

The summary of flights does _not_ require telemetry, but some pieces of it do.  E.g., your flight on 9/23 from HCR PVU HCR reads: "● Average Speed: 25.9kts ● Route distance: 41.5nm ● Longest Segment: 20.7nm"  This is all great circle.  You're not seeing this on, say, your 8/23 flight (hcr to hcr; the example you gave above seems to have been from your 8/21 flight) only has the same airport listed twice, with no telemetry, so I can't tell that you ever even left the pattern.  So the distance is simply 0.

The airports are listed in the order visited in your ROUTE, but indeed they are not in order for the METARS.  This is partly because I de-dupe.  Note that your flight here was HCR and KHCR; those didn't de-dupe.  But if you change the route to HCR PVU U14 HCR, then you'll just see 3 airports: HCR, PVU, and U14.

The reason that the spelled out name of the airport looks like a link is because it is: it re-centers the map and zooms in.  In the case of a single airport, I default to a static map (saves me money on Google maps!), but if you have telemetry, I use the (more expensive) dynamic map, and clicking the airport name zooms in on it.  I could potentially make the airport name static if the map is static.

Finally, I'm not sure what you mean by "red circle with the white double bar" - do you mean this with a single bar? 
![airport](https://user-images.githubusercontent.com/16455286/68345525-60fd3700-00a6-11ea-9700-9e8b7832c87c.png)  If so, that's an icon for an airport, trying to match what's on FAA VFR charts.

**Comment by ericberman** [comment]
Above commits address (a) not showing "No weather found" when no weather is available, and (b) not having airports linkified when the map is static. 

I **think** those are the only actionable items from this?  Did I miss any?

**Comment by KayRJay** [comment]
That's all helpful, Eric.

Without opening a new issue, I wanted to point out that my 8/21 flight summary is a bit implausible ...

● Average Speed: 498.1kts ● Path distance: 30.1nm ● Route distance: 99.6nm ● Longest Segment: 49.1nm ● Furthest points from each other: KHCR and U14 (49.1nm)

My Cirrus is fast, but not THAT fast!   I don't know what the "Path distance" is, but I do agree with the longest segment and furthest points.   What's wrong with the speed?!

**Comment by ericberman** [comment]
Completely fair question.

Average speed is computed as Math.Max(dPath, dRoute) / time, where "dpath" is the distance along the path (i.e., the actual distance you flew along whatever path through space you flew, if available from telemetry), "droute" is the distance along the route (airport-to-airport based on route, regardless of the path you flew) and time is either total logged time or (if available) flight start to flight end.

For the flight in question:
 - dpath is 30.09nm
 - dRoute is 99.611
 - time is 0.199 (logged time is 1.6, but your flight time is 0.199)

So 99.611/0.199 = approximately 500kts.  the math is entirely correct.

The problem arises from the fact that your flight start to flight end is only 0.2 hours, but your logged time is 1.6.  I use the 0.2 because that's the subset of your total time that would be "in flight" (vs, say, taxiing).   I'm not sure why you excluded so much of your flight from flight start to flight end, but that's the issue right there.

If you're using one of the apps to autodetect, it will set "Flight Start" to your first takeoff and "Flight End" to your last landing (regardless of intermediate settings; see issue #408), and the average speed will be much more reasonable.  But if you're excluding all of that time from the flight time for the flight, then there's nothing I can do but let that shrunken denominator inflate the resulting average speed.
